### PR TITLE
Notebook jest updates 

### PR DESCRIPTION
**Pull request by ps48**
### Description

1. Addresses notebook support issues with observability visualizations.
2. Updates and enhances existing tests.
3. Boosts test coverage for notebook components to reach 68%.
4. Resolves lint errors in the notebook codebase.

```
public/components/notebooks/components                   |   68.53 |    56.66 |    61.7 |   68.54 |
  note_table.tsx                                          |   95.89 |    88.88 |   93.33 |   95.89 | 137-138,329
  notebook.tsx                                            |   61.92 |    48.61 |   53.15 |   61.87 | ...1054-1060,1104
 public/components/notebooks/components/helpers           |   82.81 |    77.19 |   91.66 |   82.72 |
  default_parser.tsx                                      |   95.34 |     87.5 |     100 |   95.34 | 23,75
  legacy_route_helpers.ts                                 |     100 |      100 |     100 |     100 |
  modal_containers.tsx                                    |     100 |      100 |     100 |     100 |
  reporting_context_menu_helper.tsx                       |   66.66 |    66.66 |   76.92 |   66.25 | ...3-48,78,80,100
  zeppelin_parser.tsx                                     |   91.83 |    83.33 |     100 |   91.83 | 87-91
 ...components/notebooks/components/helpers/custom_modals |     100 |      100 |     100 |     100 |
  custom_input_modal.tsx                                  |     100 |      100 |     100 |     100 |
  reporting_loading_modal.tsx                             |     100 |      100 |     100 |     100 |
 .../components/notebooks/components/paragraph_components |   71.19 |    69.81 |   54.28 |   71.79 |
  para_input.tsx                                          |   54.34 |    56.25 |   56.25 |   55.81 | ...44,173-199,220
  para_output.tsx                                         |   98.21 |    84.61 |   83.33 |   98.21 | 134
  para_query_grid.tsx                                     |   69.69 |    18.18 |   33.33 |   73.33 | ...4,68,71-72,108
  paragraphs.tsx                                          |   64.81 |    77.35 |   55.55 |    63.8 | ...48-376,411,456
  ```

### Issues Resolved
https://github.com/opensearch-project/dashboards-observability/issues/1338

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ps48** [comment]
Waiting for https://github.com/opensearch-project/dashboards-observability/pull/1312 to be backported to main

File: public/components/notebooks/components/__tests__/notebook.test.tsx [hunk 1]
```diff
@@ -63,27 +76,290 @@ describe('<Notebook /> spec', () => {
...
+    act(() => {
+      fireEvent.input(utils.getByPlaceholderText(codePlaceholderText), {
+        target: { value: '%md \\n hello' },
+      });
+      fireEvent.click(utils.getByText('Run'));
+    });
+
+    await waitFor(() => {
+      expect(utils.queryByText('Run')).toBeNull();
+      expect(utils.getByText('hello')).toBeInTheDocument();
```
Comment (Swiddis) [inline public/components/notebooks/components/__tests__/notebook.test.tsx:157]: This is a lot of asserts/different things being tested for one test. I'd like to see more atomic tests. Other tests in this file are similar.

File: public/components/notebooks/components/notebook.tsx [hunk 2]
```diff
@@ -537,9 +543,9 @@ export class Notebook extends Component<NotebookProps, NotebookState> {
     }
   };
 
-  runForAllParagraphs = (reducer: (para: ParaType, index: number) => Promise<any>) => {
+  runForAllParagraphs = (reducer: (para: ParaType, _index: number) => Promise<any>) => {
     return this.state.parsedPara
-      .map((para: ParaType, index: number) => () => reducer(para, index))
+      .map((para: ParaType, _index: number) => () => reducer(para, _index))
```
Comment (Swiddis) [inline public/components/notebooks/components/notebook.tsx:548]: Is there a reason we update _index to indicate it's unused, while using it in reducer?

File: public/components/notebooks/components/__tests__/note_table.test.tsx [hunk 3]
```diff
@@ -23,109 +24,261 @@ jest.mock('react-router-dom', () => ({
...
+        path: 'path-1',
+        id: 'id-1',
+        dateCreated: 'date-created',
+        dateModified: 'date-modified',
+      },
+    ];
+    const { getByText, getByLabelText, queryByText } = renderNoteTable({ notebooks });
+
+    // Select a notebook
+    fireEvent.click(getByLabelText('Select this row'));
```
Comment (Swiddis) [inline public/components/notebooks/components/__tests__/note_table.test.tsx:269]: There's a lot of duplicated navigation logic here based on label text, that seems like something that's likely to break over time. Should there be a helper for this navigation, or a better selector?

File: test/sample_default_notebooks.tsx [hunk 4]
```diff
@@ -3,6 +3,13 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+export const emptyNotebook = {
```
Comment (Swiddis) [inline test/sample_default_notebooks.tsx:6]: Is there a reason these constants shouldn't also be in `notebooks_constants`?

**Review by Swiddis — APPROVED** [review]
Non-blocking issues

File: public/components/notebooks/components/notebook.tsx (hunk 2 above)
Comment (ps48) [reply public/components/notebooks/components/notebook.tsx:541]: The linter is not able to understand that index is used in reducer and leads to `no-unused-vars` error. So marked this one as an internal variable.

File: test/sample_default_notebooks.tsx (hunk 4 above)
Comment (ps48) [reply test/sample_default_notebooks.tsx]: updated. These came from legacy code tests in notebooks. Moved them to `notebooks_constants`.

File: public/components/notebooks/components/__tests__/notebook.test.tsx (hunk 1 above)
Comment (ps48) [reply public/components/notebooks/components/__tests__/notebook.test.tsx:254]: Split the tests further for code block checks.

File: public/components/notebooks/components/__tests__/note_table.test.tsx (hunk 3 above)
Comment (ps48) [reply public/components/notebooks/components/__tests__/note_table.test.tsx:269]: I'll leave this one as is for now. I'm updating some of these with cypress.

**Review by mengweieric — APPROVED** [review]

**Pull request merged**